2020-02-Spring/4_graphs/README.md

```diff
@@ -119,21 +175,170 @@ Go to [Top](#top)
 <!-- Don't remove -->
 <a name="s2"/>
 
-### 2. SOLUTION 2 TODO :bug:
+### 2. SOLUTION 2
+
+Instead of validating the status of current vertex by checking vertices(stations) it relies on, we could approach the problem top-down: we assume that all stations are shut down. Starting with known power plant thats remain active, we activate adjacent stations and substations of the current active station using DFS.Stations that are not activated by the end of the algorithm are the ones affected by shutting down K stations.
```

Comment:
```suggestion
Instead of validating the status of current vertex by checking the vertices(stations) it relies on, we could approach the problem top-down: we assume that all stations are shut down. Starting with known power plant that remains active, we activate adjacent stations and substations of the current active station using DFS. Stations that are not activated by the end of the algorithm are the ones affected by shutting down K stations.
```